src/guide/scaling-up/testing.md

```diff
@@ -49,30 +92,75 @@ Vitest is relatively new and is still undergoing rapid development. While it is
+In Vue applications, components are the main building blocks of the UI. Components are therefore the natural unit of isolation when it comes to validating your application's behavior. From a granularity perspective, component testing sits somewhere above unit testing and can be considered a form of integration testing. Much of your Vue Application should be covered by a component test and we recommend that each Vue component has its own spec file.
+
+Component tests should catch issues relating to your component's props, events, slots that it provides, styles, classes, lifecycle hooks, and more.
 
-Component tests should focus on the component's public interfaces rather than internal implementation details. Or, in other words, **test what a component does, not how it does it**.
+They should not mock child components, but instead test the interactions between your component and its children by interacting with the components as a user would. For example, a component test should click on an element like a user would instead of programmatically interacting with the component.
 
-- **DO**
+Component tests should focus on the component's public interfaces rather than internal implementation details. For most components, the public interface is limited to: events emitted, props, and slots. When testing, remember to **test what a component does, not how it does it**.
 
-  - For **Visual** tests: assert correct render output based on input props and slots.
-  - For **Interaction** tests: assert correct render updates or emitted events in response to user input events.
+**DO**
+
+- For **Visual** logic: assert correct render output based on inputted props and slots.
+- For **Behavioral** logic: assert correct render updates or emitted events in response to user input events.
+
+In the below example, we demonstrate a Stepper component that has a DOM element labeled "increment" and can be clicked. We pass a prop called `max` that prevents the Stepper from being incremented past `2`, so if we click the button 3 times, the UI should still say `2`.
+
+We know nothing about the implementation of Stepper, only that the "input" is the `max` prop and the "output" is the state of the DOM as the user will see it.
+
+```js
+const valueSelector = '[data-testid=stepper-value]'
+const buttonSelector = '[data-testid=increment]'
+
+mount(Stepper, {
+  props: {
+    max: 1
+  }
+})
+
+// Each of these commands does implicit assertions
+// that the button and values are visible
+// and that the button can be clicked.
+cy.get(valueSelector)
+  .should('be.visible')
+  .and('contain.text', '0')
+  .get(buttonSelector)
+  .click()
+  .get(valueSelector)
+  .should('contain.text', '1')
+  .click() // Should still be "1" because of the max prop
```

Comment:
I wonder if we want to at least include an example of how it looks with Test Utils (in addition), since that's something a lot of people are familiar with at this point, and most of the information you'll find on Google will reference the VTU API.

Reply:
Yeah I'm feeling weird about this, too. I actually had it written with Testing Library at first... but then I felt weird about recommending Cy, but not using Cy in the examples.

Reply:
I think this is a good opportunity to focus 'understand the concept, not the syntax'. Although not the most idiomatic Cypress, we can do something like:

___

Here's an example using Vue Test Utils, a low-level testing library that powers libraries like Testing Library and Cypress:

```js
const valueSelector = '[data-testid=stepper-value]'
const buttonSelector = '[data-testid=increment]'

const wrapper = mount(Stepper, {
  props: {
    max: 1
  }
})

// Each of these commands does implicit assertions
// that the button and values are visible
// and that the button can be clicked.
expect(wrapper.get(valueSelector).text()).toBe('0')
wrapper.get(buttonSelector).trigger('click')
expect(wrapper.get(valueSelector).text()).toBe('1')
wrapper.get(buttonSelector).trigger('click')

// Should still be "1" because of the max prop
expect(wrapper.get(valueSelector).text()).toBe('1')
```

This same test can be made more readable using Cypress; notice the ideas and concepts are the same; we are testing in a similar manner to a user, by clicking and observing the HTML. The only difference is the syntax; your testing library should not change the way you choose to test, just the syntax.

```js
mount(Stepper, {
  props: {
    max: 1
  }
})

// Each of these commands does implicit assertions
// that the button and values are visible
// and that the button can be clicked.
cy.get(valueSelector).should('contain.text', '0')
cy.get(buttonSelector).click()
cy.get(valueSelector).should('contain.text', '1')
cy.get(buttonSelector).click()

// Should still be "1" because of the max prop
cy.get(valueSelector).should('contain.text', '1'))
```

Far from perfect but this is what I was thinking. What do you think? It's a bit wordy and verbose, but I like the focus on ideas, not syntax and specific library APIs.

Reply:
@lmiller1990 exactly described my opinion which I mentioned one comment above. As I said - we can even go further and throw away everything, except a test runner (`vitest`) to help this example to be more focused